src/core/src/lib/utils.ts

```diff
@@ -10,63 +10,50 @@ export function getFieldId(formId: string, field: FormlyFieldConfig, index: stri
   return [formId, type, field.key, index].join('_');
 }
 
-export function getKeyPath(field: FormlyFieldConfigCache): (string | number)[] {
+export function getKeyPath(field: FormlyFieldConfigCache): string[] {
   if (!field.key) {
     return [];
   }
 
   /* We store the keyPath in the field for performance reasons. This function will be called frequently. */
   if (!field._keyPath || field._keyPath.key !== field.key) {
-    const path: (string | number)[] = [];
-    if (typeof field.key === 'number') {
-      path.push(field.key);
-    } else {
-      const key = field.key.indexOf('[') === -1
-        ? field.key
-        : field.key.replace(/\[(\w+)\]/g, '.$1');
-
-      (key.indexOf('.') !== -1 ? key.split('.') : [key])
-        .forEach(v => path.push(/^\d+$/.test(v) ? Number(v) : v));
-    }
-    field._keyPath = { key: field.key, path };
+    const key = field.key.indexOf('[') === -1
+      ? field.key
+      : field.key.replace(/\[(\w+)\]/g, '.$1');
+
+    field._keyPath = { key: field.key, path: key.indexOf('.') !== -1 ? key.split('.') : [key] };
   }
 
   return field._keyPath.path.slice(0);
```

Comment:
Why do you keep this method in a utils rather then put it directly on the field object?

Reply:
make sense,
it was used in several places, not only field.